src/main.py

```diff
@@ -51,66 +52,94 @@ class CameraApp(App):
 
     STREAM_NAMES = ["rgb", "disparity", "left", "right"]
 
-    def __init__(self, service_config: EventServiceConfig, stream_every_n: int) -> None:
+    def __init__(self, service_config: EventServiceConfig) -> None:
         super().__init__()
+
         self.service_config = service_config
-        self.stream_every_n = stream_every_n
 
-        self.image_decoder = ImageDecoder()
-        self.image_subscription_tasks: list[asyncio.Task] = []
+        self.image_decoder = TurboJPEG()
+
+        self.view_name = "rgb"
+
+        self.async_tasks: list[asyncio.Task] = []
 
     def build(self):
         return Builder.load_file("res/main.kv")
 
     def on_exit_btn(self) -> None:
         """Kills the running kivy application."""
+        for task in self.tasks:
+            task.cancel()
         App.get_running_app().stop()
 
+    def update_view(self, view_name: str):
+        self.view_name = view_name
+
     async def app_func(self):
         async def run_wrapper():
             # we don't actually need to set asyncio as the lib because it is
             # the default, but it doesn't hurt to be explicit
             await self.async_run(async_lib="asyncio")
-            for task in self.image_subscription_tasks:
+            for task in self.async_tasks:
                 task.cancel()
 
+        config_list = proto_from_json_file(
+            self.service_config, EventServiceConfigList()
+        )
+
+        oak0_client: EventClient | None = None
+
+        for config in config_list.configs:
+            if config.name == "oak0":
+                oak0_client = EventClient(config)
+
+        if None in [oak0_client]:
+            raise RuntimeError(f"No {config} service config in {self.service_config}")
+
         # stream camera frames
-        self.image_subscription_tasks: list[asyncio.Task] = [
-            asyncio.create_task(self.stream_camera(view_name))
+        self.tasks: list[asyncio.Task] = [
+            asyncio.create_task(self.stream_camera(oak0_client, view_name))
             for view_name in self.STREAM_NAMES
         ]
 
-        return await asyncio.gather(run_wrapper(), *self.image_subscription_tasks)
+        return await asyncio.gather(run_wrapper(), *self.tasks)
 
     async def stream_camera(
-        self, view_name: Literal["rgb", "disparity", "left", "right"] = "rgb"
+        self,
+        oak_client: EventClient,
+        view_name: Literal["rgb", "disparity", "left", "right"] = "rgb",
     ) -> None:
         """Subscribes to the camera service and populates the tabbed panel with all 4 image streams."""
         while self.root is None:
             await asyncio.sleep(0.01)
 
-        async for _, message in EventClient(self.service_config).subscribe(
-            SubscribeRequest(
-                uri=Uri(path=f"/{view_name}"), every_n=self.stream_every_n
-            ),
-            decode=True,
+        rate = oak_client.config.subscriptions[0].every_n
```

Comment:
This line feels a little fragile, but not sure a better way to do it

```suggestion
        rate = oak_client.config.subscriptions[0].every_n
```